src/test/kotlin/org/onflow/flow/sdk/impl/FlowAccessApiImplTest.kt

```diff
@@ -443,25 +448,38 @@ class FlowAccessApiImplTest {
     }
 
     @Test
-    fun `Test subscribeExecutionDataByBlockHeight error case`() = runBlocking {
+    fun `Test subscribeExecutionDataByBlockHeight error case`() = testScope.runBlockingTest {
         val blockHeight = 100L
         val exception = RuntimeException("Test exception")
 
-        `when`(mockExecutionDataApi.subscribeExecutionDataFromStartBlockHeight(any())).thenThrow(exception)
+        `when`(mockExecutionDataApi.subscribeExecutionDataFromStartBlockHeight(any()))
+            .thenAnswer { throw exception }
 
         when (val result = flowAccessApiImpl.subscribeExecutionDataByBlockHeight(blockHeight)) {
             is FlowAccessApi.AccessApiCallResponse.Success -> {
-                fail("Expected error but got success")
+                val (_, errorChannel) = result.data
```

Comment:
@jribbink after switching the implementation to be fully non-blocking the result of the error channel is fed through FlowAccessApi.AccessApiCallResponse.Success which is not ideal. 
 
I was playing around with a few different options of how to handle this, but I think we would be forced to add a blocking step to the function, e.g. something along the lines of 
```
var receivedException: Throwable? = null
        runBlocking {
            // Try to receive from the error channel
            try {
                receivedException = errorChannel.receive()
            } catch (e: ClosedReceiveChannelException) {
                // No errors
            }
        }

        if (receivedException != null) {
            FlowAccessApi.AccessApiCallResponse.Error("Failed to subscribe execution data by block height", receivedException)
        } else {
            FlowAccessApi.AccessApiCallResponse.Success(responseChannel to errorChannel)
        }
        ```

which is not what we want to have. Let me know what you think - thanks!